Source/utils/utf8.hpp

```diff
@@ -52,6 +52,14 @@ inline bool IsTrailUtf8CodeUnit(char x)
 	return static_cast<signed char>(x) < static_cast<signed char>('\xC0');
 }
 
+/**
+ * @brief Returns the number of code units for a code point starting at *src;
+ */
+inline size_t Utf8CodePointLen(const char *src)
+{
+	return "\1\1\1\1\1\1\1\1\1\1\1\1\2\2\3\4"[static_cast<unsigned char>(*src) >> 4];

```

Comment:
![image](https://github.com/diasurgical/devilutionX/assets/14297035/7d050dc3-ce68-4139-8e1c-1c63557905a3)

Reply:
OK so:

1. String literals are a very efficient way to represent static arrays.
2. We can tell the length of a UTF-8 code point by looking at the first 4 bits of a leading code point.
3. The string is basically a lookup array for all possible values of the first 4 bits.

Reply:
Compilers don't seem to handle `constexpr` arrays as well as string literals just yet. See for example:
https://github.com/fmtlib/fmt/commit/06b20387aebdd47444b9eb3d3b3bc254251fca3b

It's just as readable in a string as it would be in an array though (not very).

Reply:
oh, i think I get it now, since I know some basic UTF8